drivers/memc/memc_bflb_bl61x.c

```diff
@@ -0,0 +1,396 @@
+	/* 64 bytes */
+	tmp |= 5 << PSRAM_REG_WB_BURST_LENGTH_POS;
+	tmp |= 0 << PSRAM_REG_WB_FIX_LATENCY_POS;
+	/* Disable power down mode? */
+	tmp |= 1 << PSRAM_REG_WB_DPD_DIS_POS;
+	/* Partial refresh is no, full */
+	tmp |= 0 << PSRAM_REG_WB_PASR_POS;
+	/* Hybrid sleep off */
+	tmp |= 0 << PSRAM_REG_WB_HYBRID_SLP_POS;
+	/* Inpit power down off */
+	tmp |= 0 << PSRAM_REG_WB_IPD_POS;
+	/* Differential input clock */
+	tmp |= 0 << PSRAM_REG_WB_MCLK_TYPE_POS;
+	/* Linear mode disabled */
+	tmp |= 1 << PSRAM_REG_WB_LINEAR_DIS_POS;
+	/* Software reset Disabled */
+	tmp |= 0 << PSRAM_REG_WB_SW_RST_POS;
+	sys_write32(tmp, cfg->base + PSRAM_WINBOND_PSRAM_CONFIGURE_OFFSET);
+
+	/* Select CR0 reg */
+	tmp = sys_read32(cfg->base + PSRAM_CONFIGURE_OFFSET);
+	tmp &= PSRAM_REG_WB_REG_SEL_UMSK;
+	tmp |= 2 << PSRAM_REG_WB_REG_SEL_POS;
+	sys_write32(tmp, cfg->base + PSRAM_CONFIGURE_OFFSET);
+
+	/* Send configuration to PSRAM */
+	tmp = sys_read32(cfg->base + PSRAM_CONFIGURE_OFFSET);
+	tmp &= PSRAM_REG_CONFIG_W_PUSLE_UMSK;
+	tmp |= 1 << PSRAM_REG_CONFIG_W_PUSLE_POS;
+	sys_write32(tmp, cfg->base + PSRAM_CONFIGURE_OFFSET);
+
+	/* Wait for ack */
+	do {
+		tmp = sys_read32(cfg->base + PSRAM_CONFIGURE_OFFSET);
+		wait_ack--;
+	} while (!(tmp & PSRAM_STS_CONFIG_W_DONE_MSK) && wait_ack > 0);
+
+	memc_bflb_bl61x_release_psram_ctrl(dev);
+
+	/* Check it worked by reading PSRAM ID */
+	err = memc_bflb_bl61x_get_psram_reg(dev, 0, &check_dat);
+	if (err < 0) {
+		LOG_ERR("PSRAM check failed");
+		return err;
+	}
+	LOG_INF("PSRAM ID: %x", check_dat);
+
+	return 0;
+}
+
+static int memc_bflb_bl61x_init(const struct device *dev)
+{
+	struct memc_bflb_bl61x_data *data = dev->data;
+	const struct device *efuse = DEVICE_DT_GET_ONE(bflb_efuse);
+	const struct device *clock_dev =  DEVICE_DT_GET_ANY(bflb_clock_controller);
+	uint32_t dev_infos;
+	int err;
+	uint32_t psram_size, flash_size;
+
+
+	err = syscon_read_reg(efuse, EFUSE_DEV_INFOS_OFFSET, &dev_infos);
+	if (err < 0) {
+		LOG_ERR("Error: Couldn't read efuses: err: %d.\n", err);
+		return err;
+	}
+	psram_size = (dev_infos >> EFUSE_PSRAM_SIZE_POS) & EFUSE_PSRAM_SIZE_SIZE;
+	if (psram_size) {
+		switch (psram_size) {
+		case 1:
+			data->psram_size = 0x400000;
+		break;
+		case 2:
+			data->psram_size = 0x800000;
+		break;
+		case 3:
+			data->psram_size = 0x1000000;
+		break;
+		default:
+			data->psram_size = 0;
+		break;
+		}
+		LOG_INF("Built-in PSRAM Present, size: 0x%x bytes", data->psram_size);
+	} else {
+		LOG_INF("No Built-in PSRAM");
+	}
+	flash_size = (dev_infos >> EFUSE_FLASH_SIZE_POS) & EFUSE_FLASH_SIZE_SIZE;
+	if (flash_size) {
+		switch (flash_size) {
+		case 1:
+			data->flash_size = 0x200000;
+		break;
+		case 2:
+			data->flash_size = 0x400000;
+		break;
+		case 3:
+			data->flash_size = 0x600000;
+		break;
+		case 4:
+			data->flash_size = 0x800000;
+		break;
+		default:
+			data->flash_size = 0;
+		break;
+		}
+		LOG_INF("Built-in Flash Present, size: 0x%x bytes", data->flash_size);
```

Comment:
`flash_size` seems to only be used for logging, is it how it is supposed to be? Not sure if this information is useful

Reply:
Yes, those are for information purpose, it had no cost to add.